src/test/java/org/cactoos/list/LimitedIterableTest.java

```diff
@@ -0,0 +1,67 @@
+package org.cactoos.list;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LimitedIterable}
+ *
+ * @author Dusan Rychnovsky (dusan.rychnovsky@gmail.com)
+ * @since 0.4
+ */
+public class LimitedIterableTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void failsIfGivenLimitIsLessThanZero() {
+       new LimitedIterable<>(new ArrayAsIterable<>(), -1);
```

Comment:
Hi @yegor256, 

From the documetation of Guava - `throws IllegalArgumentException if limitSize is negative` (https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/Iterables.html#limit-java.lang.Iterable-int-).

I believe this is the correct approach. If someone is trying to get a negative number of elements from an iterable, there is most probably a bug in the code somewhere before. We want to fail fast. Not swallow the error and return some pseudo-correct result.

WDYT?